### PR TITLE
Add support for embedded resources media

### PR DESCRIPTION
**Pull request by orzech85**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
feature

### :arrow_heading_down: What is the current behavior?
Embedded resources media are not supported

### :new: What is the new behavior (if this is a feature change)?
Support to embedded resources media added

### :boom: Does this PR introduce a breaking change?
no

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines 
- [ ] Relevant documentation was updated
- [x] Rebased onto current develop

### CODE REVIEW

File: MediaManager/IMediaManager.cs [hunk 1]
```diff
@@ -50,6 +51,14 @@ public interface IMediaManager : IPlaybackManager, IDisposable
         /// <returns></returns>
         Task<IMediaItem> Play(string uri);
 
+        /// <summary>
+        /// Plays an embeded resource
+        /// </summary>
+        /// <param name="resourceName"></param>
+        /// <param name="assembly"></param>
+        /// <returns></returns>
+        Task<IMediaItem> Play(string resourceName, Assembly assembly);
```
Comment (martijn00) [inline MediaManager/IMediaManager.cs]: Assembly assembly = null

File: MediaManager/Media/IMediaExtractor.cs [hunk 2]
```diff
@@ -11,6 +12,8 @@ public interface IMediaExtractor
 
         Task<IMediaItem> CreateMediaItem(string url);
 
+        Task<IMediaItem> CreateMediaItem(string resourceName, Assembly assembly);
```
Comment (martijn00) [inline MediaManager/Media/IMediaExtractor.cs]: Assembly assembly = null

File: MediaManager/Media/MediaExtractorBase.cs [hunk 3]
```diff
@@ -14,6 +17,39 @@ public virtual Task<IMediaItem> CreateMediaItem(string url)
             return UpdateMediaItem(mediaItem);
         }
 
+        public virtual async Task<IMediaItem> CreateMediaItem(string resourceName, Assembly assembly)
```
Comment (martijn00) [inline MediaManager/Media/MediaExtractorBase.cs]: Assembly assembly = null

File: MediaManager/Media/MediaExtractorBase.cs [hunk 4]
```diff
@@ -14,6 +17,39 @@ public virtual Task<IMediaItem> CreateMediaItem(string url)
...
+        {
+            string path = null;
```
Comment (martijn00) [inline MediaManager/Media/MediaExtractorBase.cs]: if(assembly is null)
AppDomain.GetEntry()

File: MediaManager/Media/MediaLocation.cs [hunk 5]
```diff
@@ -2,7 +2,7 @@
 {
     public enum MediaLocation
     {
-        Unknown,
+        Unknown = 0,
```
Comment (martijn00) [inline MediaManager/Media/MediaLocation.cs]: Do all need to have a value?

File: MediaManager/MediaManagerBase.cs [hunk 6]
```diff
@@ -144,6 +145,15 @@ public virtual async Task<IMediaItem> Play(string uri)
             return mediaItem;
         }
 
+        public virtual async Task<IMediaItem> Play(string resourceName, Assembly assembly)
```
Comment (martijn00) [inline MediaManager/MediaManagerBase.cs]: Assembly assembly = null

File: MediaManager/Media/MediaLocation.cs (hunk 5 above)
Comment (orzech85) [reply MediaManager/Media/MediaLocation.cs]: I did that to be sure that `Unknown` is the default value.

File: MediaManager/IMediaManager.cs (hunk 1 above)
Comment (orzech85) [reply MediaManager/IMediaManager.cs]: Added `TryFindAssembly` in `MediaExtractorBase`

File: MediaManager/Media/IMediaExtractor.cs (hunk 2 above)
Comment (orzech85) [reply MediaManager/Media/IMediaExtractor.cs]: Added `TryFindAssembly` in `MediaExtractorBase`

File: MediaManager/Media/MediaExtractorBase.cs (hunk 3 above)
Comment (orzech85) [reply MediaManager/Media/MediaExtractorBase.cs]: Added `TryFindAssembly` in `MediaExtractorBase`

File: MediaManager/Media/MediaExtractorBase.cs (hunk 4 above)
Comment (orzech85) [reply MediaManager/Media/MediaExtractorBase.cs]: Added `TryFindAssembly`.

File: MediaManager/MediaManagerBase.cs (hunk 6 above)
Comment (orzech85) [reply MediaManager/MediaManagerBase.cs]: Added `TryFindAssembly` in `MediaExtractorBase`

**Pull request merged**